internal/backend/fs.go

```diff
@@ -336,16 +336,14 @@ func (s *storageFS) PatchObject(bucketName, objectName string, attrsToUpdate Obj
 	return s.CreateObject(obj, NoConditions{})
 }
 
-func (s *storageFS) UpdateObject(bucketName, objectName string, metadata map[string]string) (StreamingObject, error) {
+func (s *storageFS) UpdateObject(bucketName, objectName string, attrsToUpdate ObjectAttrs) (StreamingObject, error) {
 	obj, err := s.GetObject(bucketName, objectName)
 	if err != nil {
 		return StreamingObject{}, err
 	}
 	defer obj.Close()
-	obj.Metadata = map[string]string{}
```

Comment:
Since this is supposed to reset Metadata, we should do something like:

```go
if attrsToUpdate.Metadata != nil {
    obj.Metadata = map[string]string{}
}
```

Reply:
Ah yes that makes sense, since `obj.patch()` never resets metadata.